src/Build/Evaluation/IMetadataTable.cs

```diff
@@ -14,19 +12,19 @@ internal interface IMetadataTable
         /// Retrieves any value we have in our metadata table for the metadata name specified.
         /// If no value is available, returns empty string.
         /// </summary>
-        string GetEscapedValue(string name);
+        string? GetEscapedValue(string name);
```

Comment:
Per comments and the implementation in `EvaluatorMetadataTable`, these methods never return null. Did you have to make it return a nullable type because not all implementations are null-annotated?

Reply:
Good question. I'll switch back to this branch in the next day or so and check. Taking a quick look at the implementations of this interface they all seem to `?? string.Empty`, so I can't see why it's a nullable return.

The docs for the methods below also say `itemType` can be null, so I'll check that too.

Reply:
Took a proper look and yes, it's only `GetEscapedValueIfPresent` that can return null. And both `itemType` parameters may be null too. Updated.